### PR TITLE
SCF for operation in Tile dialect and its lowering

### PR DESCRIPTION
**Pull request by shiyao00**

### CODE REVIEW

**Review by jbruestle — COMMENTED** [review]
Great work!  The tests for the dealloc pass were fantastic, they made it very clear what the pass was supposed to be doing and covered the important cases.  Some tiny inline comments, on style/documentation.  I'll let Frank review as well, but looks good.

File: pmlc/conversion/tile_to_pxa/tests/pack_index.mlir [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// RUN: pmlc-opt -convert-tile-to-pxa %s | FileCheck %s
```
[the remark this answers is not in the document]
Comment (shiyao00) [reply pmlc/conversion/tile_to_pxa/tests/pack_index.mlir:2]: Added comment

File: pmlc/conversion/tile_to_pxa/tile_to_pxa.cc [hunk 2]
```diff
@@ -1937,6 +1926,30 @@ struct UnpackOpConversion : public OpConversionPattern<stdx::UnpackOp> {
...
 };
 
+struct ScfForOpConversion : public OpConversionPattern<scf::ForOp> {
+  using OpConversionPattern<scf::ForOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(scf::ForOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+    auto &oldBodyOps = op.getBody()->getOperations();
+    auto newOp =
```
[the remark this answers is not in the document]
Comment (shiyao00) [reply pmlc/conversion/tile_to_pxa/tile_to_pxa.cc]: Changed.

**Review by jbruestle — APPROVED** [review]

File: pmlc/conversion/tile_to_pxa/tile_to_pxa.cc [hunk 3]
```diff
@@ -179,16 +182,14 @@ struct AlwaysTrue : Matcher {
   bool match(Operation *op) const final { return true; }
 };
 
-template <typename InnerPredicate>
-struct ResultIs : Matcher {
+template <typename InnerPredicate> struct ResultIs : Matcher {
```
Comment (flaub) [inline pmlc/conversion/tile_to_pxa/tile_to_pxa.cc:185]: I'm wondering why these format changes were made? Did you use `clang-format`?

Comment (shiyao00) [reply pmlc/conversion/tile_to_pxa/tile_to_pxa.cc:185]: Yes. That's the result of my clang-format.

Comment (flaub) [reply pmlc/conversion/tile_to_pxa/tile_to_pxa.cc:185]: We use `clang-format-9`, which version were you using?

Comment (shiyao00) [reply pmlc/conversion/tile_to_pxa/tile_to_pxa.cc:185]: It's clang-format version 6.0.0. I'll update my version.

**Pull request merged**